Enable proxy cache in CI

slaperche-scality opened the pull request:
**Component**:

CI

**Context**: 

In order to reduce flakiness due to network issues, we setup the proxy-cache

**Summary**:

Setup proxy-cache for every image we use.

**Acceptance criteria**: 

Hopefully less flaky…

---

Closes: #1844
Closes: #1868

slaperche-scality committed:
ci: use proxy-cache to reduce flakiness

Refs: #1844
Refs: #1868
Signed-off-by: Sylvain Laperche <sylvain.laperche@scality.com>

slaperche-scality committed:
ci/images: install sudo & add eve to sudoers

To setup the proxy-cache we need to run commands as root, but at that
point we have already dropped back to `eve` user so let's install
`sudo` and configure it to allows `eve`.

Refs: #1844
Refs: #1868
Signed-off-by: Sylvain Laperche <sylvain.laperche@scality.com>

bert-e commented:
# Hello slaperche-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @sayf-eddine-scality

* @thomasdanan

* @carlito-scality

* @teddyandrieux

* @nootal

* @lucieleonard

* @gdemonet

* @jbertran

* @nicolast

* @monpote

* @tcarmet

* @slaperche-scality

charlesprost reviewed (changes requested):
@slaperche-scality don't we want to target dev/2.0 for this?

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @sayf-eddine-scality

* @thomasdanan

* @carlito-scality

* @teddyandrieux

* @nootal

* @lucieleonard

* @gdemonet

* @jbertran

* @nicolast

* @monpote

* @tcarmet

* @slaperche-scality





The following reviewers are expecting changes from the author, or must review again:

* @nootal

slaperche-scality committed:
ci: workaround a CI bug

The CI only watch the content of the `context` directory to know when to
rebuild the images, so if you have a Dockerfile out of the context
directory and that you only modify the Dockerfile (that's what we do in
this branch for the `lint` and `unit-test` images), then you images
won't be rebuilt.

A fix will be deployed soon, meanwhile we work around this behavior by
adding a dummy comment in `go.mod` (which is in the `context`).

See EVE-1098

Signed-off-by: Sylvain Laperche <sylvain.laperche@scality.com>

slaperche-scality commented:
> @slaperche-scality don't we want to target dev/2.0 for this?

Ho right!

Will close this one and reopen one targeting 2.0
